src/statusBar.ts

```diff
@@ -54,15 +55,19 @@ export class StatusBar extends Widget implements IStatusBar {
         let rootLayout = (this.layout = new PanelLayout());
 
         let leftPanel = (this._leftSide = new Panel());
+        let middlePanel = (this._middlePanel = new Panel());
         let rightPanel = (this._rightSide = new Panel());
 
         leftPanel.addClass(STATUS_BAR_SIDE_CLASS);
         leftPanel.addClass(STATUS_BAR_LEFT_SIDE_CLASS);
 
+        middlePanel.addClass(STATUS_BAR_MIDDLE_PANEL_CLASS);
+
```

Comment:
you should also add a `STATUS_BAR_SIDE_CLASS ` to the middle area, possibly rename it to something agnostic of sides. Also check the corresponding CSS to make sure that there isn't anything specific to right/left.